### PR TITLE
generate pseudo-random username for each 2fa test

### PR DESCRIPTION
**Pull request by jeremywmoore**
2fa tests were all using the same test user which seems to occasionally cause issues related to unintended otp rate-limiting. This PR passes a login_data fixture to each test with a pseudo-random username to improve isolation between the tests and avoid accidental rate-limiting.

### CODE REVIEW

File: tests/test_2fa.py [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
 
 import pyotp
+import pytest
 from auth_helper import register_user_2fa
 from flask.testing import FlaskClient
 
+LoginData: TypeAlias = dict[str, str]
+
+@pytest.fixture()
+def nonce(len: int = 7) -> str:
```
Comment (rmlibre) [inline tests/test_2fa.py:14]: Let's bump up the nonce length just to make sure there's no duplicates. I'd recommend 16.

File: tests/test_2fa.py [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
+  return ''.join(random.choices(string.ascii_lowercase +  # noqa: S311
+                             string.digits, k=len))
+
+@pytest.fixture()
+def login_data(nonce: str) -> LoginData:
+  return {"username": "test_user" + nonce, "password": "SecurePassword123!"}
```
Comment (rmlibre) [inline tests/test_2fa.py:20]: For readability, let's add an underscore between the static name part & the nonce.

```python
"test_user_" + nonce
```

**Review by rmlibre — CHANGES_REQUESTED** [review]
Looks good. Thanks!

File: tests/test_2fa.py (hunk 1 above)
Comment (jeremywmoore) [reply tests/test_2fa.py]: addressed by [894a61d](https://github.com/scidsg/hushline/pull/460/commits/894a61d0a16545e7ce55fbe6dd980f6e1aa45903)

File: tests/test_2fa.py (hunk 2 above)
Comment (jeremywmoore) [reply tests/test_2fa.py]: addressed by [89fc892](https://github.com/scidsg/hushline/pull/460/commits/89fc892b238cacd35022fa4ab2f19235c2576f1e)

**Comment by jeremywmoore** [comment]
FYI I extended the nonce to 15 characters instead of 16 due to a max username length of 25.

**Review by rmlibre — APPROVED** [review]
👍🏾

**Review by micahflee — APPROVED** [review]

**Pull request merged**